majilove/models.py

```diff
@@ -106,3 +108,402 @@ def display_blood_type(self):
+    message_translation = models.TextField(_('Message translation'), max_length=500, null=True)
+    MESSAGE_TRANSLATIONs_CHOICES = ALL_ALT_LANGUAGES
+    d_message_translations = models.TextField(_('Message translation'), null=True)
+    @property
+    def t_message_translation(self):
+        if get_language() == 'ja': return None
+        return self.message_translations.get(get_language(), self.message_translation)
+
+    # photo stats
+
+    #triplets_in_moments is number of sets of +30 dance/vocal/charm sqaures before 100%
+    RARITIES = OrderedDict([
+        (1, {
+            'short_form': 'N',
+            'special_shot_percentage': None,
+            'outfit_unlock_percentage': 0,
+            'triplets_in_moments' : 1,
+            'max_levels': 20,
+            }),
+        (2, {
+            'short_form': 'R',
+            'special_shot_percentage': 100,
+            'outfit_unlock_percentage': 50,
+            'triplets_in_moments' : 4,
+            'max_levels': (30, 50),
+            }),
+        (3, {
+            'short_form': 'SR',
+            'special_shot_percentage': 83,
+            'outfit_unlock_percentage': 33,
+            'triplets_in_moments' : 6,
+            'max_levels': (40, 60),
+            }),
+        (4, {
+            'short_form': 'UR',
+            'special_shot_percentage': 87,
+            'outfit_unlock_percentage': 25,
+            'triplets_in_moments': 8,
+            'max_levels': (50, 70),
+        }),
+    ])
+
+    RARITY_CHOICES = [(_name, _info['short_form']) for _name, _info in RARITIES.items()]
+    RARITY_WITHOUT_I_CHOICES = True
+    i_rarity = models.PositiveIntegerField(_('Rarity'), choices=i_choices(RARITY_CHOICES), db_index=True)
+
+    COMBINABLE_RARITIES = [2, 3, 4]
+
+    @property
+    def combinable(self):
+        return self.i_rarity in self.COMBINABLE_RARITIES
+
+    @property
+    def single_max_level(self):
+        return RARITIES[self.i_rarity]['max_levels'][0] if self.combinable else RARITIES[self.i_rarity]['max_levels']
+
+    @property
+    def max_max_level(self):
+        return RARITIES[self.i_rarity]['max_levels'][1] if self.combinable else RARITIES[self.i_rarity]['max_levels']
+
+    COLORS= OrderedDict([
+        (1, { # yellow
+            'translation': _('Star'),
+            'english': u'Star'
+        }),
+        (2, { # red
+            'translation': _('Shine'),
+            'english': u'Shine'
+        }),
+        (3, { # blue
+            'translation': _('Dream'),
+            'english': u'Dream'
+        })
+    ])
+
+    COLOR_CHOICES = [(_name, _info['translation']) for _name, _info in COLORS.items()]
+    COLOR_WITHOUT_I_CHOICES=True
+    i_color = models.PositiveIntegerField(_('Color'), choices=COLOR_CHOICES, db_index=True)
+    english_color = property(getInfoFromChoices('color', COLORS, 'english'))
+
+    STATS = OrderedDict([
+        (1, {
+            'translation': _('Dance'),
+            'english': u'Dance',
```

Comment:
The third stat is "Charm" in the worldwide version and "ACT" in the Japanese version of the game. It feels worth keeping that difference in the code rather than just the translations, especially because it's not intuitive.